### PR TITLE
JMS 2.0 features added to transport receiver

### PR DESCRIPTION
**Pull request by erandacr**

### CODE REVIEW

File: modules/jms/src/main/java/org/apache/axis2/transport/jms/ServiceTaskManagerFactory.java [hunk 1]
```diff
@@ -343,18 +345,19 @@ private static int getCacheLevel(Map<String,String> svcMap, Map<String,String> c
...
-        if (val == null || "1.1".equals(val)) {
-            return true;
+        if (val == null || JMSConstants.JMS_SPEC_VERSION_1_1.equals(val)) {
+            return JMSConstants.JMS_SPEC_VERSION_1_1;
+        } else if (JMSConstants.JMS_SPEC_VERSION_2_0.equals(val)) {
+            return JMSConstants.JMS_SPEC_VERSION_2_0;
         } else {
-            return false;
-        }
+            return JMSConstants.JMS_SPEC_VERSION_1_0;        }
```
Comment (virajsenevirathne) [inline modules/jms/src/main/java/org/apache/axis2/transport/jms/ServiceTaskManagerFactory.java]: Formatting Issue

File: modules/jms/src/main/java/org/apache/axis2/transport/jms/JMSListener.java [hunk 2]
```diff
@@ -151,7 +151,7 @@ private boolean checkJMSConnection(ServiceTaskManager stm) {
                     jmsConFactory,
                     jmsProperties.get(JMSConstants.PARAM_JMS_USERNAME),
                     jmsProperties.get(JMSConstants.PARAM_JMS_PASSWORD),
-                    stm.isJmsSpec11(), null, false, "");
+                    stm.getJmsSpec(), null, false, "");
```
Comment (VijithaEkanayake) [inline modules/jms/src/main/java/org/apache/axis2/transport/jms/JMSListener.java]: Better to define constant value for EMPTY_STRING("") and use it.

File: modules/jms/src/main/java/org/apache/axis2/transport/jms/JMSMessageReceiver.java [hunk 3]
```diff
@@ -192,6 +194,16 @@ private boolean processThoughEngine(Message message, UserTransaction ut)
                     contentTypeInfo.getPropertyName()));
         }
 
+        // Setting JMSXDeliveryCount header on the message context
+        try {
+            int deliveryCount = message.getIntProperty(JMS_MESSAGE_DELIVERY_COUNT_HEADER);
+            msgContext.setProperty(JMSConstants.DELIVERY_COUNT, deliveryCount);
+        } catch (NumberFormatException nfe) {
+            if (log.isDebugEnabled()) {
+                log.debug("JMSXDeliveryCount is not set in the received message");
```
Comment (isururanawaka) [inline modules/jms/src/main/java/org/apache/axis2/transport/jms/JMSMessageReceiver.java]: Exception info only prints id debug logs are enabled

File: modules/jms/src/main/java/org/apache/axis2/transport/jms/JMSListener.java (hunk 2 above)
Comment (erandacr) [inline modules/jms/src/main/java/org/apache/axis2/transport/jms/JMSListener.java]: Fixed with https://github.com/wso2/wso2-axis2-transports/pull/106

**Pull request merged**